src/test/java/com/exasol/spark/BaseIntegrationSetup.java

```diff
@@ -0,0 +1,110 @@
+package com.exasol.spark;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.sql.SparkSession;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.testcontainers.junit.jupiter.Container;
+
+import com.exasol.containers.ExasolContainer;
+import com.exasol.containers.ExasolDockerImageReference;
+import com.exasol.dbbuilder.dialects.exasol.ExasolObjectFactory;
+import com.exasol.dbbuilder.dialects.exasol.ExasolSchema;
+
+/**
+ * A base integration test class with Exasol docker container setup.
+ */
+public class BaseIntegrationSetup {
+    private static final Logger LOGGER = Logger.getLogger(BaseIntegrationSetup.class.getName());
+    private static final String DEFAULT_DOCKER_IMAGE = "7.1.19";
+
+    @Container
+    private static final ExasolContainer<? extends ExasolContainer<?>> EXASOL = new ExasolContainer<>(
+            getExasolDockerImage()).withReuse(true);
+
+    protected static Connection connection;
+    protected static ExasolObjectFactory factory;
+    protected static ExasolSchema exasolDatabase;
+    protected static SparkSession spark;
+
+    @BeforeAll
+    public static void beforeAll() throws SQLException {
+        EXASOL.purgeDatabase();
+        connection = EXASOL.createConnection();
+        factory = new ExasolObjectFactory(connection);
+        spark = SparkSessionProvider.getSparkSession(createSparkConfiguration());
+        createDatabase("DEFAULT_SCHEMA");
+    }
+
+    @AfterAll
+    public static void afterAll() throws SQLException {
+        dropDatabase();
+        connection.close();
+        spark.close();
+    }
+
+    private static void createDatabase(final String exasolSchemaName) {
+        LOGGER.fine(() -> "Creating a new schema '" + exasolSchemaName + '"');
+        dropDatabase();
+        exasolDatabase = factory.createSchema(exasolSchemaName);
+    }
+
+    private static void dropDatabase() {
+        if (exasolDatabase != null) {
+            LOGGER.fine(() -> "Dropping schema '" + exasolDatabase.getName() + '"');
+            exasolDatabase.drop();
+            exasolDatabase = null;
+        }
+    }
+
+    public Map<String, String> getOptionsMap() {
+        final Map<String, String> map = Stream.of(new String[][] { //
+                { "host", EXASOL.getDockerNetworkInternalIpAddress() }, //
+                { "port", EXASOL.getDefaultInternalDatabasePort() + "" }, //
+                { "username", EXASOL.getUsername() }, //
+                { "password", EXASOL.getPassword() }, //
+                { "jdbc_url", EXASOL.getJdbcUrl() }, //
+        }).collect(Collectors.toMap(e -> e[0], e -> e[1]));
```

Comment:
Using `Map.of()` would be easier to read ;)

Reply:
Ahh, yes again Java 8 :(